astropy/coordinates/baseframe.py

```diff
@@ -576,12 +612,43 @@ def represent_as(self, new_representation, in_frame_units=False):
         <SkyCoord (ICRS): (x, y, z) [dimensionless]
             ( 1.,  0.,  0.)>
         """
-        new_representation = _get_repr_cls(new_representation)
 
-        cached_repr = self.cache['representation'].get((new_representation.__name__,
-                                                        in_frame_units))
+        n_diffs = len(self.data.differentials)
+        has_diffs = n_diffs > 0
+
+        if isinstance(new_representation, six.string_types):
+            new_representation = _get_repr_cls(new_representation)
+
+        if new_differential:
+            if not has_diffs:
+                raise TypeError('Frame data has no associated differentials '
+                                '(i.e. the frame has no velocity data) - '
+                                'represent_as() only accepts a new '
+                                'representation class.')
+
+            if isinstance(new_differential, six.string_types):
```

Comment:
Here it is needed because `new_differential` might be a `dict`. That is, when we pass this to `BaseRepresentation.represent_as`, it has to be either a class or a `dict`